### PR TITLE
Fix collateral selection

### PR DESCRIPTION
**Pull request by dfar-sga**
This addresses some bug reports regarding collateral selection. The changes are as follows:
- try to account for minUTxO value during selection
- reduce minimum collateral to 3 ADA
- try to select collateral from all wallet UTxOs if initial selection fails
- change `beforeCip30Methods` to `false` in `defaultSynchronizationParams` to prevent chain query while signing (addresses `Could not get utxo` error)

### CODE REVIEW

File: src/Contract/Config.purs [hunk 1]
```diff
@@ -140,7 +140,10 @@ defaultTimeParams =
 defaultSynchronizationParams :: ContractSynchronizationParams
 defaultSynchronizationParams =
   { syncBackendWithWallet:
-      { errorOnTimeout: false, beforeCip30Methods: true, beforeBalancing: true }
+      { errorOnTimeout: false
```
Comment (klntsky) [inline src/Contract/Config.purs:143]: Why this change?

Comment (dfar-sga) [reply src/Contract/Config.purs:143]: This is specific to Liqwid projects, our infrastructure doesn't currently index far enough into the past and we have frequent bug reports from wallet UTxO queries failing.

As an aside, there's similar reasoning for reducing the required collateral to 3 ADA, which is based on the current observed fees for Liqwid's transactions. Should this be configurable?
